zerver/lib/upload.py

```diff
@@ -130,6 +130,16 @@ def get_realm_for_filename(path):
         return None
     return get_user_profile_by_id(key.metadata["user_profile_id"]).realm.id
 
+def delete_message_image_s3(path_id):
```

Comment:
OK; it should be possible for you to setup the S3 integration in development to test it.  The docs in zproject/local_settings_template.py should I think explain what you need to do, except that the secrets file is `zproject/dev-secrets.conf` and you may need to tweak the not case of the `if PRODUCTION` block in zproject/settings.poy.